.azure/scripts/build.sh

```diff
@@ -12,21 +12,19 @@ echo "Source branch: ${BRANCH}"
 make build
 
 if [ "$BUILD_REASON" == "PullRequest" ] ; then
-  make docker_build
+  sh ./docker-images/build-images.sh build
   echo "Building Pull Request - nothing to push"
 elif [[ "$BRANCH" != "refs/tags/"* ]] && [ "$BRANCH" != "refs/heads/main" ]; then
-  make docker_build
+    sh ./docker-images/build-images.sh build
   echo "Not in main branch or in release tag - nothing to push"
 else
   if [ "$BRANCH" == "refs/heads/main" ]; then
       export DOCKER_TAG="latest"
   else
       export DOCKER_TAG="${BRANCH#refs/tags/}"
   fi
-
-  make docker_build
```

Comment:
I think this logic is wrong. The container images should be built in every build. But only selected builds should push them. Here (and I assume also in the changed version) you build only the Java code in every build and the container images only in the selected cases. That means that you might not detect many issues in PR builds for example.

Reply:
Yeah that makes sense -> I guess that will be enough to move the `build` part outside this `if/else` block and only if the branch will be `main` or `release`, images will be pushed. Btw inside the `./docker-images/build-images.sh build` Java code is build (by mvn command) and also `docker_build` is executed. Did you mean something else that we don't check ATM?

Reply:
Well, ideally, you wanna do `make build` and `make docker_build` everytime. And then do `make docker_push` (not `make all`) separately only on the right branches.

Reply:
Okay, so I'll change it to do just `push` and not `make all`
Thanks :)